conans/client/toolchain/cmake/ios.py

```diff
@@ -0,0 +1,97 @@
+import textwrap
+
+from .base import CMakeToolchainBase
+
+
+class CMakeiOSToolchain(CMakeToolchainBase):
+    _toolchain_tpl = textwrap.dedent("""
+        {% extends 'base_toolchain' %}
+        {% block before_try_compile %}
+            {{ super() }}
+            # set cmake vars
+            set(CMAKE_SYSTEM_NAME {{ CMAKE_SYSTEM_NAME }})
+            set(CMAKE_SYSTEM_VERSION {{ CMAKE_SYSTEM_VERSION }})
+            set(DEPLOYMENT_TARGET ${CONAN_SETTINGS_HOST_MIN_OS_VERSION})
+            # Set the architectures for which to build.
+            set(CMAKE_OSX_ARCHITECTURES {{ CMAKE_OSX_ARCHITECTURES }})
+            # Setting CMAKE_OSX_SYSROOT SDK, when using Xcode generator the name is enough
+            # but full path is necessary for others
+            set(CMAKE_OSX_SYSROOT {{ CMAKE_OSX_SYSROOT }})
+            if(NOT DEFINED CMAKE_XCODE_ATTRIBUTE_DEVELOPMENT_TEAM)
+              set(CMAKE_XCODE_ATTRIBUTE_DEVELOPMENT_TEAM "123456789A" CACHE INTERNAL "")
+            endif()
+        {% endblock %}
+        {% block main %}
+            {{ super() }}
+            {% if shared_libs -%}
+            message(STATUS "Conan toolchain: Setting BUILD_SHARED_LIBS= {{ shared_libs }}")
+            set(BUILD_SHARED_LIBS {{ shared_libs }})
```

Comment:
This is already part of the ``CMakeGenericToolchain``, can't be reused? This is what I struggle to see with templates.

Reply:
It would be inherited... if this toolchain was inheriting from the CMakeGenericToolchain, but it isn't.

If we are sure that these values are going to be used by all the toolchains, then they can (and have to) be moved to the `CMakeToolchainBase`.